### PR TITLE
[SHIMGVW] Display the file name in taskbar even if the file isn't loaded

### PR DESCRIPTION
**Pull request by 3ts75**
## Purpose

In win2k3, the file name was displayed in the title bar even if the file was not loaded.
However, in React OS, the file name was not displayed in the tie and river when the file was not loaded.
Therefore, the file name was displayed unless the file name could't to retrieved.

JIRA issue: [CORE-18954](https://jira.reactos.org/browse/CORE-18954)

### CODE REVIEW

File: dll/win32/shimgvw/shimgvw.c [hunk 1]
```diff
@@ -524,9 +524,10 @@ pLoadImageFromNode(SHIMGVW_FILENODE *node, HWND hwnd)
     pLoadImage(node->FileName);
 
     LoadStringW(hInstance, IDS_APPTITLE, szResStr, _countof(szResStr));
-    if (image != NULL)
+
+    pchFileTitle = PathFindFileNameW(node->FileName);
+    if (pchFileTitle != NULL && wcsicmp(pchFileTitle, L"") != 0)
```
Comment (KRosUser2) [inline dll/win32/shimgvw/shimgvw.c:529]: is "!= 0" really needed here ?

Comment (3ts75) [reply dll/win32/shimgvw/shimgvw.c:529]: This isn't necessary, fix it.

File: dll/win32/shimgvw/shimgvw.c [hunk 2]
```diff
@@ -507,7 +507,7 @@ pLoadImageFromNode(SHIMGVW_FILENODE *node, HWND hwnd)
 {
     WCHAR szTitleBuf[800];
     WCHAR szResStr[512];
-    LPWSTR pchFileTitle;
+    LPWSTR pchFileTitle = NULL;
```
Comment (JoachimHenze) [inline dll/win32/shimgvw/shimgvw.c:510]: Is this change really needed? As far as I can see there is no access to it before the assignment via PathFindFileNameW().
And https://learn.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-pathfindfilenamew
seems to promise to write something there in all flow-cases.

Comment (JoachimHenze) [inline dll/win32/shimgvw/shimgvw.c:510]:
```suggestion
    LPWSTR pchFileTitle;
```

File: dll/win32/shimgvw/shimgvw.c [hunk 3]
```diff
@@ -524,9 +524,10 @@ pLoadImageFromNode(SHIMGVW_FILENODE *node, HWND hwnd)
     pLoadImage(node->FileName);
 
     LoadStringW(hInstance, IDS_APPTITLE, szResStr, _countof(szResStr));
-    if (image != NULL)
+
+    pchFileTitle = PathFindFileNameW(node->FileName);
+    if (pchFileTitle != NULL && wcsicmp(pchFileTitle, L""))
```
Comment (JoachimHenze) [inline dll/win32/shimgvw/shimgvw.c:529]:
```suggestion
    if (pchFileTitle && wcsicmp(pchFileTitle, L"")) 
```

File: dll/win32/shimgvw/shimgvw.c (hunk 2 above)
Comment (3ts75) [reply dll/win32/shimgvw/shimgvw.c:510]: This isn’t necessary, fix it.

File: dll/win32/shimgvw/shimgvw.c (hunk 3 above)
Comment (3ts75) [reply dll/win32/shimgvw/shimgvw.c:529]: This isn’t necessary, fix it.

Comment (HBelusca) [reply dll/win32/shimgvw/shimgvw.c]: What part is not necessary?
Also, the construct `wcsicmp(pchFileTitle, L"")` can be more efficiently replaced by `*pchFileTitle` .

Comment (3ts75) [reply dll/win32/shimgvw/shimgvw.c]: "!= NULL" isn't necessary.

Change `wcsicmp(pchFileTitle, L"")` to `*pchFileTitle`.

**Review by JoachimHenze — APPROVED** [review]
Thank you very much for your patch and your interest in the project.
Congratulation to your first contributions approval, Yukinari Mitsu!
I just approved your proposal to run on our buildbots.

**Review by HBelusca — COMMENTED** [review]
Suggested changes in the title and description of the main commit:
```diff
-[SHIMGVW] Displayed file name in task bar if file not loaded
+[SHIMGVW] Display the file name in taskbar even if the file isn't loaded
+CORE-18954

-If file not loaded, the file name wasn't displayed in the taskbar.
+If the file could not be loaded, its name wasn't displayed in the taskbar.
-so, if not loaded, displayed file name in taskbar.
+Now, always display the file name in all cases.
```

**Comment by JoachimHenze** [comment]
> Suggested changes in the title and description of the main commit:

Contributor does not need to act, we can do that when merging.

**Review by HBelusca — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]

**Pull request merged**